unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs

```diff
@@ -346,5 +367,22 @@ public RightOfData(VisualElement element, long id)
                 this.element = element;
             }
         }
+
+        /*
+         * Calculates what web browsers call "device pixel ratio": a ratio between the application virtual resolution
+         * and the screen resolution. It represents how many physical/real pixels are needed to draw the virtual pixels
+         * of the application. The smaller the application resolution is, compared to the real screen resolution, the
+         * higher the device pixel ratio is. Normally the result can be between 1~3.
+         * Note: Explorer's 'rendering scale' doesn't affect this, as that is only for 3D rendering, so it doesn't affect
+         * the whole application resolution (e.g. UIs are unaffected).
+        */
+        private float CalculateDevicePixelRatio()
+        {
+            // 'Screen.mainWindowDisplayInfo' cannot be used in WebGL due to 'Screen.GetDisplayInfoForPoint' unsupported error
+            float realScreenDiagonalSizeInPixels = Mathf.Sqrt((Screen.currentResolution.width * Screen.currentResolution.width) + (Screen.currentResolution.height * Screen.currentResolution.height));
+            float viewportDiagonalSizeInPixels = Mathf.Sqrt((Screen.width * Screen.width) + (Screen.height * Screen.height));
+
+            return realScreenDiagonalSizeInPixels / viewportDiagonalSizeInPixels;
+        }
```

Comment:
is it the same as doing it this way? (one less sqrt)
```suggestion
        private float CalculateDevicePixelRatio()
        {
            // 'Screen.mainWindowDisplayInfo' cannot be used in WebGL due to 'Screen.GetDisplayInfoForPoint' unsupported error
            float realScreenDiagonalSizeInPixels = (Screen.currentResolution.width * Screen.currentResolution.width) + (Screen.currentResolution.height * Screen.currentResolution.height);
            float viewportDiagonalSizeInPixels = (Screen.width * Screen.width) + (Screen.height * Screen.height);

            return Mathf.Sqrt(realScreenDiagonalSizeInPixels / viewportDiagonalSizeInPixels);
        }
```

Reply:
Uhmm, you are right, mathematically speaking that should be the same, let me check

Reply:
This entire function can be simplified to `return Screen.width / Screen.currentResolution.width;`

Reply:
menduz has a point there, if we are only handling landscape screen orientation, it could be simplified to just the width

Reply:
Done! https://github.com/decentraland/unity-renderer/pull/5314/commits/a0180b5baf126e99927edd056b7b8d0515975624